lib/garage.rb

```diff
@@ -0,0 +1,34 @@
+class Garage
+
+  attr_reader :lot
+
+  def initialize
+    @lot = []
+  end
+
+  def add_car(car)
+    @lot << car
+  end
+
+  def total_cars
+    @lot.count
+  end
+
+  def total_cars_by_make(make)
+    @lot.find_all {|car| car.make == make}.count
+  end
+
+  def paint_all_cars_one_color(color_desired)
+    @lot.map do |car|
```

Comment:
Question for you: why are you using `.map` here? I can see an argument for it, but don't think it's necessary. Would be interested in hearing your reasoning one way or another.